src/System.Net.Http/tests/FunctionalTests/HttpClient.SelectedSitesTest.cs

```diff
@@ -0,0 +1,128 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Threading.Tasks;
+
+using Xunit;
+
+namespace System.Net.Http.Functional.Tests
+{
+    public abstract class HttpClient_SelectedSites_Test : HttpClientTestBase
+    {
+        [Theory]
+        [OuterLoop]
+        [Trait("SelectedSites", "true")]
+        [MemberData(nameof(GetSelectedSites))]
+        public async Task RetrieveSite_Succeeds(string site)
+        {
+            // Not doing this in bulk for platform handlers.
+            if (!UseSocketsHttpHandler)
+                return;
+
+            int remainingAttempts = 2;
+            while (remainingAttempts-- > 0)
+            {
+                try
+                {
+                    await VisitSite(site);
+                    return;
+                }
+                catch
+                {
+                    if (remainingAttempts < 1)
+                        throw;
+                    await Task.Delay(1500);
+                }
+            }
+
+            throw new Exception("Not expected to reach here");
+        }
+
+        [Theory]
+        [OuterLoop]
+        [Trait("SiteInvestigation", "true")]
+        [InlineData("http://microsoft.com")]
+        public async Task RetrieveSite_Debug_Helper(string site)
+        {
+            await VisitSite(site);
+        }
+
+        public static IEnumerable<string[]> GetSelectedSites()
+        {
+            const string resourceName = "SelectedSitesTest.txt";
```

Comment:
It's a relatively short list... how about just putting it into an array in the code?

Reply:
With this I can easily swap to much larger list without changing the sources - that's what I've been doing in my runs (~1500 sites).

Reply:
Ok